core/math/vector3.cpp

```diff
@@ -149,7 +145,17 @@ Basis Vector3::to_diagonal_matrix() const {
 			0, 0, z);
 }
 
-Vector3::operator String() const {
+Vector3 Vector3::clamped(real_t p_len) const {
+	real_t l = length();
+	Vector3 v = *this;
+	if (l > 0 && p_len < l) {
+		v /= l;
+		v *= p_len;
+	}
 
+	return v;
+}
```

Comment:
This can probably be done more efficiently, e.g. (untested) :
```
	Vector3 v = *this;

	real_t sl = length_squared();
	if (sl > 0)
	{
		real_t p_len_squared = p_len * p_len;

		if (sl < p_len_squared)
		{
			real_t l = Math::sqrt(sl);

			v *= (p_len / l);
		}
	}

	return v;
```
You don't have to do the test against `p_len_squared`, you can do the sqrt before that, depending on the expected data.

Noteworthy, that you can speed up the case where length is zero (by avoiding the sqrt), and you can combine the vector division and multiply.

Reply:
Looks good! I'll update the one in Vector2 as well.

Reply:
I better profile it first just in case hehe! :smile:

Reply:
Yeah it is an improvement, is 3x faster for me on some random test data.